app.py

```diff
@@ -13,8 +14,22 @@ def generate_weights(edges):
     return weights
 
 
-def q1(g, w):
-    r = g.V[random.randint(0, len(g.V) - 1)]  # pick random start
+def max_with_inf(list):
+    maximum = float('-inf')
+    for item in list:
+        if item > maximum and item !=float('inf'):
+            maximum = item
+    return maximum
```

Comment:
can be done with:
max([max([x for x in a if x!=float('inf')]) for a in w])

Reply:
done